CHANGELOG.md

```diff
@@ -20,6 +20,7 @@
 * [CHANGE] Store-gateway: the deprecated configuration parameter `-blocks-storage.bucket-store.consistency-delay` has been removed. #5050
 * [CHANGE] The configuration parameter `-blocks-storage.bucket-store.bucket-index.enabled` has been deprecated and will be removed in Mimir 2.11. Mimir is running by default with the bucket index enabled since version 2.0, and starting from the version 2.11 it will not be possible to disable it. #5051
 * [FEATURE] Query-frontend: add `-query-frontend.log-query-request-headers` to enable logging of request headers in query logs. #5030
+* [FEATURE] Store-gateway: retain lazy-loaded index headers between restarts by eagerly loading them during startup. #5606
```

Comment:
[nit] We should explain that this feature is experimental, disabled by default and only supported if lazy loading is enabled.